rickchurch/models.py

```diff
@@ -1,52 +1,35 @@
-import enum
-from typing import NamedTuple, Optional
-
-from fastapi import HTTPException
-
-
-class AuthState(enum.Enum):
-    """Represents possible outcomes of a user attempting to authorize."""
-    NO_TOKEN = (
-        "There is no token provided, provide one in an Authorization header in the format "
-        "'Bearer {your token here}' or go to church's home and get one."
-    )
-    BAD_HEADER = "The Authorization header does not specify the Bearer scheme."
-    INVALID_TOKEN = "The token provided is not a valid token or has expired."
-    BANNED = "You are banned."
-    MODERATOR = "This token belongs to a moderator."
-    USER = "This token belongs to a regular user."
-
-    def __bool__(self) -> bool:
-        """Return whether the authorization was successful."""
-        return self in (AuthState.USER, AuthState.MODERATOR)
-
-    def raise_if_failed(self) -> None:
-        """Raise an HTTPException if a user isn't authorized."""
-        if self:
-            return
-        raise HTTPException(status_code=403, detail=self.value)
-
-    def raise_unless_mod(self) -> None:
-        """Raise an HTTPException if a moderator isn't authorized."""
-        if self == AuthState.MODERATOR:
-            return
-        elif self == AuthState.USER:
-            raise HTTPException(status_code=403, detail="This endpoint is limited to moderators")
-        self.raise_if_failed()
-
-
-class AuthResult(NamedTuple):
-    """The possible outcomes of authorization with user id."""
-    state: AuthState
-    user_id: Optional[int]
-
-    def __bool__(self) -> bool:
-        """Return whether the authorization was successful"""
-        return bool(self.state)
-
-    def raise_if_failed(self) -> None:
-        """Raise an HTTPException if a user isn't authorized."""
-        self.state.raise_if_failed()
-
-    def raise_unless_mod(self) -> None:
-        self.state.raise_unless_mod()
+import re
+
+import pydantic
+
+_RGB_RE = re.compile(r"[0-9a-fA-F]{6}")
+
+
+class Task(pydantic.BaseModel):
+    """A task used by the API."""
+    x: int
+    y: int
+    rgb: str
+
+    # Validators for x and y aren't added, because we don't know canvas dimensions
```

Comment:
Could we make a method to better validate the `x,y` provided?

Reply:
I don't think it's possible without a blocking function. Since we'd need to make a request to the `get_size` pixels API without async. The only way it could be done is if we stored the dimesnions, but that would mean whenever they change, we would have to either hard-code the new values, or it could probably be possible to get those dynamically, but it would still require a reset of the whole church.

The good news is that Tasks are generated by us, not by the users, they only submit them, so we can just check that the submitted task matches and the check isn't really that important

Reply:
Ill approve it, and eventually we could go about adding in a method to validate the `x,y` coordinates as other people might try and use this and manage to crash the program.